### DISCUSSION TITLE
[RFC] support pre-compiled mainline kernels

### DISCUSSION THREAD

**Discussion by arighi** [arighi/virtme-ng#37] [Ideas]
TL;DR use pre-compiled mainline kernels from Ubuntu mainline builds repository (https://kernel.ubuntu.com/mainline) to completely cut off build time when we want to run specific upstream kernel versions.

See: https://github.com/arighi/virtme-ng/pull/35

Feedbacks are welcome (especially for the command line syntax used to select these pre-compiled kernels).

**Comment by arighi** [comment]
I've done a lot of testing with this new feature and it seems to work really well, it is also very very helpful, I've already used it to do a fast bisect that saved me a lot of intermediate builds.

So, I decided to go ahead a merge this today. I'm not yet 100% convinced about the syntax (specifying tags like `vng -r v6.6-rc3`), maybe we should have used something like `vng -r https:v6.6-rc3` to make it clear that these kernels are downloaded from an external website, but if it becomes a problem we can adjust the syntax in the future, hopefully without breaking any existent workflow.

Anyway, this new feature will be available in the next version of virtme-ng! :partying_face:

**Comment by andreaterzolo** [comment]
Hi! Thank you for the feature! I tried to use it recently, the first time for an issue with kernel 6.11.4 and the second one today for an issue with kernel 6.12.1. Looking at https://kernel.ubuntu.com/mainline/v6.12.1/ it seems like there are pre-built kernels only for `s390x`  while I'm running on `amd64`. The same issue is true for the latest 6.11 patch versions.

I maintain an ebpf probe and usually, I need to understand which is the first kernel version that introduced the offending change, so having a lot of pre-built kernels could really speed up the triaging. The issue is that these are missing for the latest kernel versions... do you have any idea how we can improve this?
